### ISSUE TITLE
Set Frame for Revolute/Prismatic Joints

### ISSUE THREAD

**Issue by dmklee** [dimforge/rapier.js#271]
I would like to create joints where the joint frame is translated and rotated w.r.t. the parent body frame.  Similar to how joints can be defined in a [URDF](http://wiki.ros.org/urdf/XML/joint).  

Fixed joints already allow this, and [this PR](https://github.com/dimforge/rapier/pull/473#issue-1640918244) extends the capability for spherical joints in rust.  I tried modifying Generic Joint to accept arguments for `frame1` and `frame2` but I was not able to get it to work.

**Comment by EthanHermsey** [comment]
"There is a similar request to allow for different axis in this issue."

That's surreal.. It opens an issue I created! I came here through a google search.. Still trying to get multiple axis going... ;p

**Comment by wkwooten** [comment]
Yeah this was a major roadblock for my project for a while... Fortunately, I was able to get it resolved so I'll submit a PR shortly.

**Comment by EthanHermsey** [comment]
Oh my! That was the only reason I couldn't migrate from ammo to rapier.
Thank you very much!
